src/nvim/option.c

```diff
@@ -196,6 +196,7 @@ static void set_init_default_shell(void)
     } else {
       set_string_default(kOptShell, (char *)shell, false);
```

Comment:
if `_noalloc` were used above, then could do this this here:
```suggestion
      set_string_default(kOptShell, (char *)shell, true);
```

Reply:
Technically you are right, `set_string_default()` could change accordingly. 

However, in case we do that, we run into the following problem.

1. In one of the `if` branche we apply `malloc()` for `kOptShell` defaults
2. In the other `if` branch we use statically allocated memory

This becomes an issue when the `free_all_options() -> optval_free()` calls are applied. While 1) would require `xfree()`, 2) would be throwing an error if `xfree()` applied at this point...